src/yaml/mapping.cr

```diff
@@ -96,7 +96,19 @@ module YAML
       {% end %}
     {% end %}
 
-    def initialize(%pull : ::YAML::PullParser)
+    def self.new(%pull : ::YAML::PullParser)
+      if obj = %pull.read_alias(\{{@type}})
+        return obj
+      end
+
+      instance = allocate
```

Comment:
Shouldn't `instance` be a fresh variable? Similarly `obj` above.

Reply:
Not necessarily. There's no risk of clashing other variables.